core/domain/user_id_migration.py

```diff
@@ -516,3 +516,269 @@ class ModelsUserIdsHaveUserSettingsExplorationsVerificationJob(
     def entity_classes_to_map_over(cls):
         """Return a list of datastore class references to map over."""
         return SEPARATE_MODEL_CLASSES
+
+
+class AddAllUserIdsVerificationJob(jobs.BaseMapReduceOneOffJobManager):
+    """For every rights model merge the data from all the user id fields
+    together and put them in the all_user_ids field of an appropriate
+    RightsAllUsersModel.
+    """
+
+    @staticmethod
+    def _add_collection_user_ids(rights_model, all_users_model):
+        """Compare the existing CollectionRightsAllUsersModel with the user IDs
+        in the CollectionRightsModel, if some of the user IDs from snapshots are
+        not in the parent rights model return them.
```

Comment:
Seems like you do more than that -- you change the datastore too? It's important to explain this, because at first I assumed this was just a GET method...

Shouldn't we report an error and then look into it?

Ditto below.

Reply:
Done. We return them and then set `SUCCESS-NOT_SUBSET` as the result so we can analyze this after the migration.